src/main/java/teammates/ui/webapi/action/GetStudentAction.java

```diff
@@ -17,37 +25,56 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
-        Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
-        switch (intent) {
-        case STUDENT_SUBMISSION:
-            String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-            String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
-            FeedbackSessionAttributes feedbackSession = logic.getFeedbackSession(feedbackSessionName, courseId);
-            StudentAttributes studentAttributes = getStudentOfCourseFromRequest(feedbackSession.getCourseId());
-            checkAccessControlForStudentFeedbackSubmission(studentAttributes, feedbackSession);
-            break;
-        case FULL_DETAIL:
-            // TODO implement this when necessary
-            break;
-        default:
-            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        CourseAttributes course = logic.getCourse(courseId);
+        StudentAttributes student = null;
+
+        if (userInfo.isInstructor) {
```

Comment:
I think we should not branch the authentication based on the type of login entity. What if a student want to get the information of himself while he is also an instructor?
Therefore, we should branch based on the parameters provided:
- `key` -> do nothing as long as the key is correct
- `courseId` -> student should be in the course
- `courseId` + `email` -> authenticate instructor

Reply:
Yeap your right! I noticed this issue when I was testing locally as well (I was masquerading as a student while instructor).

I will update accordingly! (need to refactor some of the tests also after this)